src/magic/model/MagicAmount.java

```diff
@@ -10,17 +10,35 @@ public int getAmount(final MagicEvent event) {
         return getAmount(event.getSource(), event.getPlayer());
     }
 
+    public int getPositiveAmount(final MagicSource source, final MagicPlayer player) {
+        if (this == MagicAmountFactory.NegXCost) {
```

Comment:
this looks very unintuitive, for getPositiveAmount to return a negative value in a special case

Reply:
NegXCost is to handle something like pay {X} and target creature gets -X/-X

Reply:
Recommend removing the special case